nginx_stage/lib/nginx_stage/user.rb

```diff
@@ -33,21 +33,32 @@ class User
     # @param user [String] the user name defining this object
     # @raise [ArgumentError] if user or primary group doesn't exist on local system
     def initialize(user)
-      @passwd = Etc.getpwnam user.to_s
-      @group = Etc.getgrgid gid
-      @groups = get_groups
-
-      if name.to_s != user.to_s
-        err_msg = <<~HEREDOC
-          Username '#{user}' is being mapped to '#{name}' in SSSD and they don't match.
-          Users with domain names cannot be mapped correctly. If '#{name}' still has the
-          domain in it you'll need to set SSSD's full_name_format to '%1$s'.
+      # "123user".to_i.to_s (123) != "123user".to_s
+      # "user123".to_i.to_s (0) != "user123".to_s
+      # "user".to_i.to_s (0) != "user".to_s
+      # "123".to_i.to_s (123) == "123".to_s
+      if user.to_i.to_s == user.to_s
+        # The user is composed of all numbers, (numeric string), test if it is a uid
+        @passwd = Etc.getpwuid(user.to_i) rescue nil
```

Comment:
I'm about to leave for the day, but I thought I'd leave this comment.

This seems OK to me on the surface (without thinking about it so hard), but `rescue` blocks are actually pretty expensive because of the stack management they have to do. I'm not too concerned about the runtime here, but I know you mentioned avoiding a regex because of it's overhead - which is to say I'd be willing to bet a rescue block has more overhead, but I'd have to actually benchmark it to be sure.

Just food for thought really, not an actual request to change anything.

Reply:
So I benchmarked the various scenarios:

Regexp is 8% faster than string conversion in the if statement (we're still talking 0.1-0.2 microseconds difference, I did run it 1M times and did this several platforms, it is consistently ~8-10% slower to do to_i.to_s == to_s vs a 'regexp')

Then the call to Env (getpwuid) is "slow" at 10 microseconds for a valid cached user or 13 microseconds for getpwnam, however spawning a shell to call id is 100x slower - for 1000 calls, the shell method takes 2.01s, vs the Etc method which takes 0.024s.

IF the call goes to the rescue (which only happens if a user happens to be all-digit username but is not actually a UID, what documentation I can find, apparently valid in RHEL6 and 7 but cannot be created from RHEL8 onwards, it IS slow at 200 microseconds to go through the rescue. But I would find it unusual people have logins with all-digit usernames, but it can happen.

Likewise the call to rescue in the group is only hit when a group is still attached to the user, but the group itself has been deleted which is generally a transient issue. We don't want to fail because of that.

Compared to the ~25 milliseconds it takes to resolve a user/group from LDAP through SSSD...